Architect-Coders/MU8
# Issue #11: Configure Detekt

**Configure Detekt:** Issue #11 .

- Detekt configured and added to all modules.
- Improved domain module and updated app to presentation.

**Testing Done:**

* On Emulator: Nexus 4 API 27.

**How it looks:**

* In Android Studio:

![detekt-evidence-1](https://user-images.githubusercontent.com/3532556/69484635-8c508780-0e03-11ea-8293-d720f5e05145.jpg)

* In terminal:

![detekt-evidence-2](https://user-images.githubusercontent.com/3532556/69484636-8c508780-0e03-11ea-8c18-ecaf90c32723.jpg)

**How to test:**

* Scenario 0: Configure detekt on your AS, to do this reach the modulo from Architect Coders where this is mentioned below the video: https://members.architectcoders.com/topic/1-5-reglas-para-deteccion-de-problemas/. Once you have the plugin installed, go to the settings and check the options like this:

![image](https://user-images.githubusercontent.com/3532556/69484669-20225380-0e04-11ea-83ba-2aa9298ee31c.png)

* To test this configuration, try to put a `;` in a Kotlin class like `MainActivity` and you'll see that the auto correct will remove it for you.
* Scenario 1: On the `MainActivity` create a new function that has a parameter with the first capital letter, check that you get marked in red that you need to declare it in lower case. 
* Scenario 2: Keeping the previous change, go to the terminal and run the command `./gradle detekt` on Mac/Linux or `gradlew detekt` on Windows. You'll see that after it has checked it will show a build failed error with the message of where it failed as in the screenshot above.

**Signed-off-by:**

* Diego Recalde <<difereto89@gmail.com>>
